### PR TITLE
Fix issues identified by sonarlint

### PR DESCRIPTION
**Pull request by kingthorin**
Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
@ricekot @psiinon @thc202 could I get input from any of you on this? (Shouldn't be too contentious, but a second look is always helpful.)

File: src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java [hunk 1]
```diff
@@ -143,7 +144,9 @@ private MarkItemColorHighlighter getHighligher() {
                     getHistoryExtension()
                             .getHistoryReferencesTable()
                             .getModel()
-                            .getColumnIndex(DefaultHistoryReferencesTableModel.Column.HREF_ID);
+                            .getColumnIndex(
+                                    org.zaproxy.zap.view.table.HistoryReferencesTableModel.Column
```
Comment (thc202) [inline src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java:148]: Could import the class..

File: src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java [hunk 2]
```diff
@@ -62,33 +62,34 @@
...
-        // CORPORATE EDITION
-        new Color(0xe0ffff),
-        new Color(0xa8c0c0),
-        new Color(0x708080),
-        new Color(0x384040),
-        // Placeholder
-        PLACEHOLDER
-    };
+    static List<Color> palette =
+            Arrays.asList(
```
Comment (thc202) [inline src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java:66]: This returns a fixed size list, attempting to add more elements will lead to an exception. You need to use other implementation (i.e. pass it into an `ArrayList`).

**Comment by kingthorin** [comment]
Thanks @thc202!

File: src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java (hunk 2 above)
Comment (kingthorin) [reply src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java]: Tweaked, let me know what you think.

Comment (thc202) [reply src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java]: Looks good.

Comment (kingthorin) [reply src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java]: Thanks!

Comment (thc202) [reply src/main/java/org/zaproxy/zap/extension/neonmarker/ExtensionNeonmarker.java]: No problem.

**Pull request merged**